src/ngraph/op/reverse_sequence.cpp

```diff
@@ -41,21 +42,23 @@ void op::ReverseSequence::validate_and_infer_types()
     auto input_shape = get_input_partial_shape(0);
     auto input_rank = input_shape.rank();
 
-    NODE_VALIDATION_CHECK(this,
-                          input_rank.is_dynamic() || m_batch_axis < size_t(input_rank),
-                          "Batch axis index (",
-                          m_batch_axis,
-                          ") is out of bounds (argument shape: ",
-                          input_shape,
-                          ").");
-
-    NODE_VALIDATION_CHECK(this,
-                          input_rank.is_dynamic() || m_seq_axis < size_t(input_rank),
-                          "Sequence axis index (",
-                          m_seq_axis,
-                          ") is out of bounds (argument shape: ",
-                          input_shape,
-                          ").");
+    if (m_batch_axis < 0 || m_seq_axis < 0)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              input_rank.is_static(),
+                              "In order to handle negative axes input_rank must be static (",
+                              "batch_axis=",
+                              m_batch_axis,
+                              ", seq_axis=",
+                              m_seq_axis,
+                              ")");
+    }
+
+    if (input_rank.is_static())
+    {
+        m_batch_axis = ngraph::normalize_axis(this, m_batch_axis, static_cast<int64_t>(input_rank));
```

Comment:
Have separate members `m_batch_normalized_axis` and `m_seq_normalized_axis` and set them. That way if the inputs change rank and validation gets re-run, the right thing happens. It also means the serialized graph will have the provided attribute values as opposed to the normalized attributes.

Reply:
Done, but I am sure to all aspects of you review. I added additional methods to serialize origin axes values. `get_batch_axis` and `get_sequence_axis` methods return normalized axes values in order to leave `size_t` type expected by current backends.